entrypoint.sh

```diff
@@ -0,0 +1,17 @@
+#!/bin/sh -l
+
+# download relevant branch and write the diff to a local file
+diff_filename="./nightfalldlp_raw_diff.txt"
+if [ "$GITHUB_BASE_REF" ]; then
+  git fetch origin "$GITHUB_BASE_REF" --depth=1;
+  echo "PR: fetching diff between origin/$GITHUB_BASE_REF and $GITHUB_SHA";
+  git diff origin/"$GITHUB_BASE_REF" "$GITHUB_SHA" > $diff_filename;
+else
+  git fetch origin "$EVENT_BEFORE" --depth=1;
+  echo "PUSH: fetching diff between $EVENT_BEFORE and $GITHUB_SHA";
+  git diff "$EVENT_BEFORE" "$GITHUB_SHA" > $diff_filename;
```

Comment:
How come we don't need to have lines like `ARG GITHUB_BASE_REF` for these Github env vars? Are they just injected into the container by Github?

Reply:
yep exactly

Reply:
Do you think we need to just check that they are there when we build the image, with `RUN test -n "$GITHUB_BASE_REF"`. Or is this excessive?

Reply:
Do these git actions also work for private repos?

Reply:
I would think if by this point we have the Env variable inside the running environment (Dockerfile above, which kickstarts this), git command would work (similar to how once we setup the correct SSH on our computer, we can run git command without reauthenticate).

It is a good thing to test for sure - once this is merge, we can test that on the test repo.

Reply:
We don't have access to these variables in the Dockerfile, and since Github is populating them I'm not sure what we would really get from validating them.

These git commands should work for private repos but we will need to test to confirm